src/nvim/eval.c

```diff
@@ -12195,18 +12194,23 @@ static int list2proftime(typval_T *arg, proftime_T *tm) FUNC_ATTR_NONNULL_ALL
   }
 
   int error = false;
-  varnumber_T n1 = list_find_nr(arg->vval.v_list, 0L, &error);
-  varnumber_T n2 = list_find_nr(arg->vval.v_list, 1L, &error);
+  int32_t high = list_find_nr(arg->vval.v_list, 0L, &error);
+  int32_t low = list_find_nr(arg->vval.v_list, 1L, &error);
   if (error) {
     return FAIL;
   }
 
+  // When we converted the proftime_T to two 32-bit ints, we shifted the high
+  // part by one and made the sign bit of the low part 0. Undo these changes.
+  low = (high & 1 ? low + INT32_MAX + 1 : low);
```

Comment:
It does seem that the C standard leaves this undefined, although assuming that all supported platforms use two's compliment, that shouldn't be a problem. Still, changing `high` and `low` to `uint32_t`s, makes it well defined, and I think your solution looks much simpler. :+1: 

I'll check that the same change works in `f_reltime()`, too.